Source/WebCore/html/PDFDocument.cpp

```diff
@@ -207,16 +207,16 @@ void PDFDocument::postMessageToIframe(const String& name, JSC::JSObject* data)
 
 void PDFDocument::sendPDFArrayBuffer()
 {
-    auto arrayBuffer = loader()->mainResourceData()->tryCreateArrayBuffer();
-    if (!arrayBuffer) {
-        ASSERT_NOT_REACHED();
-        return;
+    if (auto* documentLoader = loader()) {
+        if (auto mainResourceData = documentLoader->mainResourceData()) {
+            if (auto arrayBuffer = mainResourceData->tryCreateArrayBuffer()) {
```

Comment:
What happens if these 2 checks fail? Do we simply show a blank page? A fallback to our usual error codepath would be nice if possible.

Reply:
I wasn't able to reproduce this condition to see for sure what happens, but I would indeed expect to see a blank page. That's probably actually correct, though, because the load was stopped.

I don't see other error paths in this file; what did you have in mind regarding "usual error codepath"?

Reply:
Well, I suppose there could be other reasons for there to be no main resource data, like a network error or something. Hopefully those are handled elsewhere? Or is that what's missing here?

Reply:
I think I'll leave this alone unless you have specific suggestions for improvement.